tool/tsh/tsh.go

```diff
@@ -1034,31 +1034,45 @@ func onListClusters(cf *CLIConf) {
 		utils.FatalError(err)
 	}
 
-	var clusters []services.RemoteCluster
+	var rootCluster services.RemoteCluster
+	var leafClusters []services.RemoteCluster
 	err = client.RetryWithRelogin(cf.Context, tc, func() error {
 		proxyClient, err := tc.ConnectToProxy(cf.Context)
 		if err != nil {
 			return err
 		}
 		defer proxyClient.Close()
 
-		clusters, err = proxyClient.GetAllClusters(cf.Context)
-		return err
+		var rootErr, leafErr error
+		rootCluster, rootErr = proxyClient.GetRootCluster()
+		leafClusters, leafErr = proxyClient.GetLeafClusters(cf.Context)
+		return trace.NewAggregate(rootErr, leafErr)
 	})
 	if err != nil {
 		utils.FatalError(err)
 	}
+
+	profile, _, err := client.Status("", cf.Proxy)
```

Comment:
`profile` can be returned as nil here even if err is nil so you prob. need to handle this scenario as well.